### DISCUSSION TITLE
Issue with FlashLoanArbitrage + Use FakeDex to simulate (able to deploy FakeDex, but unable to execute functions)

### DISCUSSION THREAD

**Discussion by Glow-in-the-dark** [Cyfrin/foundry-full-course-cu#629] [Q&A] [answered]
This is my code repo:
https://github.com/Glow-in-the-dark/Solidity_Flashloan_Hardhat

I am able to deploy on Sepolia Testnet, with:
- ( Dex.sol, address below)
> FakeDex Contract deployed:  0xB8De86Fce1BDE6e3D33CEb00C4e3F0c7D5C6752d

I am trying to deposit DAI, so that after deposit, i can then run the buyDAI() function, to buy DAI at a discount.
However, when i tried to call the `depositDAI()` and `depositUSDC()` functions
![Screenshot 2023-08-13 at 4 49 25 PM](https://github.com/smartcontractkit/full-blockchain-solidity-course-js/assets/2751458/c3736ba7-259f-4b6e-a1be-d94f92edc54b)

the seems like nothing happen. it just keep saying pending. (Not sure what's the issue)

**Comment by theirrationalone** [comment]
Checkout your metamask's transactions section..... 

Apply more gas to fulfill txs...

**Reply by Glow-in-the-dark** [reply]
mmmm... weird, there was no metamask popup to adjust the gas for the transactions tho. ( and otherwise how do i do that?)
Can you provide more details ? Thanks and appreciate it.

**Reply by Glow-in-the-dark** [reply]
i tried to call the function from etherscan after verifying it.
Metamask now popups ( if i execute it via etherscan)
<img width="321" alt="Screenshot 2023-08-14 at 2 03 44 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/3300587d-3928-4ad3-8caf-d1816a3643dd">
This is the error message.
<img width="1353" alt="Screenshot 2023-08-14 at 2 05 59 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/2751458/1cc1e1d0-5fd5-413f-b3f9-86434b395f52">

**Reply by Glow-in-the-dark** [reply, marked as answer]
Ahhh ok nvm i got it....I think it was an approval problem.
